ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpWork.java

```diff
@@ -65,11 +65,20 @@ int maxEventLimit() throws Exception {
     return maxEventLimit;
   }
 
-  void overrideEventTo(Hive fromDb) throws Exception {
+  void overrideEventTo(Hive fromDb, long bootstrapLastId) throws Exception {
     if (eventTo == null) {
       eventTo = fromDb.getMSC().getCurrentNotificationEventId().getEventId();
       LoggerFactory.getLogger(this.getClass())
           .debug("eventTo not specified, using current event id : {}", eventTo);
     }
+
+    // If we are bootstrapping ACID tables, we need to restrict events only upto the event id at
+    // the beginning of the bootstrap dump. See bootstrampDump() for more details.
+    if (bootstrapLastId >= 0) {
```

Comment:
Can we check this before we invoke getCurrentNotificationEventId and shall return there?

Reply:
That's a better optimization. Will save some cycles going to the metastore. Done.